Assets/_GameJamCat/Scripts/CatBehaviour.cs

```diff
@@ -6,16 +6,50 @@ namespace GameJamCat
 {
     public class CatBehaviour : MonoBehaviour
     {
-        // Start is called before the first frame update
-        void Start()
+        [SerializeField] private Renderer _catRenderer;
+
+        private ICatFactory _factory = null;
+
+        public ICatFactory Factory
```

Comment:
There shouldn't be a cat factory in the CatBehaviour, the factory should only be inside the CatManager since the manager should have access/handles the list of cats.

Reply:
I get that, I've always thought the best approach for this kind of thing was to keep a reference on the cloned object since that scales better to multiple factories but I guess that doesn't really apply here. I'll remove it

Reply:
hmmm how come you think it's best to keep a reference to the cloned object? 
isn't the CatBehaviour itself a cloned object?